include/RAJA/pattern/params/reducer.hpp

```diff
@@ -23,29 +25,115 @@ struct ValLoc {
   using index_type = RAJA::Index_type;
   using value_type = T;
 
-  RAJA_HOST_DEVICE ValLoc() {}
-  RAJA_HOST_DEVICE ValLoc(value_type v) : val(v) {}
-  RAJA_HOST_DEVICE ValLoc(value_type v, RAJA::Index_type l) : val(v), loc(l) {}
-
-  RAJA_HOST_DEVICE void min(value_type v, index_type l) { if (v < val) { val = v; loc = l; } }
-  RAJA_HOST_DEVICE void max(value_type v, index_type l) { if (v > val) { val = v; loc = l; } }
+  RAJA_HOST_DEVICE constexpr ValLoc() {}
+  RAJA_HOST_DEVICE constexpr explicit ValLoc(value_type v) : val(v) {}
+  RAJA_HOST_DEVICE constexpr ValLoc(value_type v, RAJA::Index_type l) : val(v), loc(l) {}
 
-  bool constexpr operator<(const ValLoc& rhs) const { return val < rhs.val; }
-  bool constexpr operator>(const ValLoc& rhs) const { return val > rhs.val; }
+  RAJA_HOST_DEVICE constexpr bool operator<(const ValLoc& rhs) const { return val < rhs.val; }
+  RAJA_HOST_DEVICE constexpr bool operator>(const ValLoc& rhs) const { return val > rhs.val; }
 
-  value_type getVal() {return val;}
-  RAJA::Index_type getLoc() {return loc;}
+  RAJA_HOST_DEVICE constexpr value_type getVal() const {return val;}
```

Comment:
If this is constexpr everything here should probably be constexpr.

Reply:
Everything here should already be `constexpr`; am I missing something in particular?

Reply:
no, looks good now.